test/release_notes.py

```diff
@@ -0,0 +1,177 @@
+    entry = rnm.ReleaseNoteEntry(
+        mimetype='text/markdown',
+        contents=message,
+        category=rnm.ReleaseNotesCategory.BUGFIX,
+        audience=rnm.ReleaseNotesAudience.DEVELOPER,
+        author=rnm.ReleaseNotesAuthor(
+            hostname='github.com',
+            username='dummyuser',
+        ),
+        pullrequest='https://github.com/example/pr/1'
+    )
+    return rnm.ReleaseNotesDoc(
+        ocm=rnm.ReleaseNotesOcmRef(
+            component_name='github.com/example/test',
+            component_version='1.2.3',
+        ),
+        release_notes=[entry],
+    )
+
+
+def test_attach_release_notes_resource():
+    component = DummyComponent()
+    oci_client = DummyOciClient()
+    image_ref = 'example.com/test/image:1.2.3'
+
+    docs = [create_dummy_doc()]
+    tarstream = rnt.release_notes_docs_into_tarstream(docs)
+    tar_bytes = b''.join(tarstream)
+    expected_digest = f'sha256:{hashlib.sha256(tar_bytes).hexdigest()}'
+
+    oci_client.put_blob(
+        image_reference=image_ref,
+        digest=expected_digest,
+        octets_count=len(tar_bytes),
+        data=tar_bytes,
+    )
+
+    component.resources.append(
+        ocm.Resource(
+            name='release-notes-archive',
+            version=component.version,
+            type=ocm.ArtefactType.BLOB,
+            relation=ocm.ResourceRelation.LOCAL,
+            access=ocm.LocalBlobAccess(
+                localReference=expected_digest,
+                size=len(tar_bytes),
+                mediaType='application/tar',
+            ),
+        )
+    )
+
+    assert len(oci_client.uploads) == 1
+
+    assert len(component.resources) == 1
+
+
+def test_collect_release_notes_merges_sources(monkeypatch):
+    component = DummyComponent()
+    dummy_component_descriptor = DummyComponentDescriptor(component)
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        repo_path = os.path.abspath(tmpdir)
+        local_notes_dir = os.path.join(repo_path, '.ocm', 'release-notes')
+        os.makedirs(local_notes_dir, exist_ok=True)
+
+        local_doc = create_dummy_doc('Local fix')
+        local_yaml_path = os.path.join(local_notes_dir, 'dummy.yaml')
+        with open(local_yaml_path, 'w') as f:
+            yaml.dump(
+                dataclasses.asdict(local_doc),
+                Dumper=ocm.EnumValueYamlDumper,
+                stream=f,
+                allow_unicode=True,
+            )
+
+        fetched_doc = create_dummy_doc('Fetched fix')
+        monkeypatch.setattr(
+            'release_notes.fetch.fetch_release_notes',
+            lambda **kwargs: [fetched_doc]
+        )
+        monkeypatch.setattr(
+            'release_notes.ocm.release_notes_range_recursive',
+            lambda **kwargs: [
+                (ocm.ComponentIdentity(name='github.com/example/dep', version='0.1.0'),
+                 'Subcomponent fix')
+            ]
+        )
+        monkeypatch.setattr(
```

Comment:
I think mocking so many functions is not a good pattern. To me, this is an indication that we should either re-organise our code (e.g. do dependency-injection where missing), extend the testing-setup, or re-organise the tests